### ISSUE TITLE
Missing XmlSlurper import in generated test classes

### ISSUE THREAD

**Issue by christian-anke** [spring-cloud/spring-cloud-contract#560]
When using the following groovy script the resulting test class cannot be run because of the missing import for the XmlSlurper.

TestContract.groovy:

``` 
org.springframework.cloud.contract.spec.Contract.Contract.make {
     request {
        method 'GET'
        url '/'
    }
    response {
        status 200
        body()
        headers {
            contentType('text/xml')
        }
    }
}
```

Error when running:

```
cannot find symbol
  symbol:   class XmlSlurper
```

**Comment by marcingrzejszczak** [comment]
Can you present the whole stack trace, the generated test, the spring cloud contract version? Also we don't really support XML :/

**Comment by trfulcher** [comment]
Is there an outline of what is missing or lacking for XML support? Fairly new to using this project but we have a lot of legacy services exposing XML that we'd like to define contracts against.

**Comment by marcingrzejszczak** [comment]
There is no converting of XML entries to XPath. So basically everything is missing :/ So what you can do however is use the WireMock proxy example (https://github.com/marcingrzejszczak/the-legacy-app#generating-tests-against-the-real-api-via-wiremock) and code here - https://github.com/marcingrzejszczak/the-legacy-app/tree/master/stubs_with_proxy . So by using this approach you can set up a proxy to access your application's API. All the requests and responses would get stored as WireMock json stubs that later will get packaged as a JAR. You can then use Stub Runner to use those on the client side.

**Comment by marcingrzejszczak** [comment]
does it answer your question @trfulcher ? can I close it?

**Comment by trfulcher** [comment]
Hi,

Sorry, on holiday right now. Yes I suppose closed is fine. I've yet to try
the proxy approach you suggest. It might be nice to understand if native
XML/xpath route was a likely enhancement if enough people rooted for it

Tim

**Comment by marcingrzejszczak** [comment]
Yeah, we could consider it. If anyone is interested then please upvote this issue

**Comment by andoniarroyo** [comment]
I have the same issue.
The generated tests are trying to instantiate the class _XmlSlurper_, but without the proper import, so it will always fail to check a response with content type 'application/xhtml+xml'
This case is not so uncommon, so I think the tool should define a way to extend the generated template...

Any thoughts?

**Comment by marcingrzejszczak** [comment]
We'll try to add some basic XML support for 2.1.0

**Comment by andoniarroyo** [comment]
Amazing mate,
Thanks for the response!

**Comment by trfulcher** [comment]
That sounds great indeed. Is there a timeline for 2.1.0 even an advance
peek - on the website I see the latest shown is 2.0.0-rc1


Tim

**Comment by marcingrzejszczak** [comment]
We don't have the timeline for 2.1.0. We want to first release 2.0.0 ;)
